File: OCH_Tao/ABC/B/272.py
N,M = map(int,input().split())
X = [set() for _ in range(N)]
for i in range(M):
  k,*x = map(int,input().split())
  for j in x:
    X[j-1].update(x)
print("Yes" if all([len(i)==N for i in X]) else "No")

File: OCH_Tao/ABC/B/272.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const [N, M] = input.shift().split(" ").map(x => parseInt(x, 10));
let X = new Array(N).fill().map(() => new Set());
for (let i = 0; i < M; i++) {
  let [, ...x] = input[i].split(" ").map(x => parseInt(x, 10));
  for (const j of x) {
    for (const k of x) {
      X[j - 1].add(k);
    }
  }
}
console.log(X.every(x => x.size === N) ? "Yes" : "No");